core/garment/src/garment.ts

```diff
@@ -895,6 +895,12 @@ async function garmentFromWorkspace(
       watch: true,
       runnerOptions
     });
+  } else {
+    if (!watcherStarted) {
+      for (const meta of metaByAction.values()) {
+        meta?.onDestroyHandler?.();
```

Comment:
I logged meta here for some batch actions and they didn't have an `onDestroyHandler`.

I put the following code in `runActionsInBatch` and it seemed to work, what do you think?:

![image](https://user-images.githubusercontent.com/34415120/106784622-6f892400-6644-11eb-86e3-14059dc38ce3.png)

![image](https://user-images.githubusercontent.com/34415120/106784773-9f382c00-6644-11eb-8b5d-0b6842296dc7.png)

Reply:
@stephan-noel I dont get why would you assign a to the batchMeta the first encountered action's meta? probably you can assign there only a batch's onDestroyHandler? Can you please push into this PR a whole solution?

Reply:
@beshanoe  you're right, I was only considering my own use case in which I need `context.longRunning` to only be called once per runner.  

I guess maybe there are several approaches here:
- Call `context.longRunning` once per runner (NOT per project per runner) like I do above.  If you want to handle shutdown for several projects, you can do something like (ignore the invocation of onetime below) :

![image](https://user-images.githubusercontent.com/34415120/107555342-f69f4480-6bce-11eb-8a40-a7b715e8e8ec.png)

- Call context.longRunning once per action (the approach you were expecting of per project per runner), and if the cleanup had already been done by a previous project (ie, the projects shared the same port and I already closed it), then I have to detect that it was already cleaned up or limit it the invocations somehow.

Honestly the first approach seemed more natural to me, but I may be biased to my use case, what are your thoughts?

Reply:
@beshanoe Changed it a bit to separate the non-batch meta from batchMeta.  I do set and get for batchMetaByAction where I do to bring attention to the fact that having one meta per action instead of per runner is a possibility in the future.  What do you think?